src/regex.nim

```diff
@@ -1566,7 +1566,7 @@ iterator peek[T: seq[Rune] | string](s: T): (int, Rune, Rune) {.inline.} =
   yield (j, prev, invalidRune)
 
 # todo: make it iterative
-proc step(
+proc step*(
```

Comment:
are you sure this must be public?

Reply:
I just tried this. The issue is in the `template stepFrom()` below. It seems the method syntax has some limitations and can't be resolved from `result.step(...)` to `step(result, ...)`, but changing that fixes the issue.

Reply:
ok, just fixed it this way ^^

Reply:
I reported the issue https://github.com/nim-lang/Nim/issues/7085